### PR TITLE
Update workflow template documentation

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Update workflow template documentation

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _automating-configurations/workflow-templates.md [hunk 1]
```diff
@@ -6,9 +6,11 @@ nav_order: 25
 
 # Workflow templates
 
-OpenSearch provides several workflow templates for some common machine learning (ML) use cases, such as semantic or conversational search. 
+OpenSearch provides several workflow templates for some common machine learning (ML) use cases. Using a template simplifies complex setups required for use cases like semantic or conversational search.
```
Comment (amitgalitz) [inline _automating-configurations/workflow-templates.md:9]: Can we explain here maybe that we also support pre defined template based on a given use case? and also user can give there own template if they want and then link to create workflow API?
Then we can give example:

File: _automating-configurations/workflow-templates.md [hunk 2]
```diff
@@ -6,9 +6,11 @@ nav_order: 25
 
 # Workflow templates
 
-OpenSearch provides several workflow templates for some common machine learning (ML) use cases, such as semantic or conversational search. 
+OpenSearch provides several workflow templates for some common machine learning (ML) use cases. Using a template simplifies complex setups required for use cases like semantic or conversational search. 
 
-You can specify a workflow template when you call the [Create Workflow API]({{site.url}}{{site.baseurl}}/automating-configurations/api/create-workflow/). To provision the workflow, specify `provision=true` as a query parameter. For example, you can configure [neural sparse search]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-search/) by providing the `local_neural_sparse_search_bi_encoder` query parameter as `use_case`, as shown in the following request:
+For example, you can configure the setup required for [neural sparse search]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-search/) by using the `local_neural_sparse_search_bi_encoder` predefined template. When you use a template, the multi-step setup that required running separate API requests is accomplished with a single request.
```
Comment (amitgalitz) [inline _automating-configurations/workflow-templates.md:11]: I am also still unsure if we should have this as the first example as the other one simplifies a lot more complexity vs this one that simplifies 3 API calls only

**Review by amitgalitz — APPROVED** [review]
Thank you for these changes!

File: _automating-configurations/workflow-templates.md [hunk 3]
```diff
@@ -6,38 +6,15 @@ nav_order: 25
 
 # Workflow templates
 
-OpenSearch provides several workflow templates for some common machine learning (ML) use cases, such as semantic or conversational search. 
+OpenSearch provides several workflow templates for some common machine learning (ML) use cases. Using a template simplifies complex setups required for use cases like semantic or conversational search.
```
Comment (dbwiddis) [inline _automating-configurations/workflow-templates.md:9]:
```suggestion
OpenSearch provides several workflow templates for some common machine learning (ML) use cases. Using a template simplifies complex setups and provides many default values for use cases like semantic or conversational search. 
```

**Review by dbwiddis — APPROVED** [review]

File: _automating-configurations/index.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ You can automate complex OpenSearch setup and preprocessing tasks by providing t
 In OpenSearch 2.12, configuration automation is limited to ML tasks.
 {: .info}
 
-OpenSearch use case templates provide a compact description of the setup process in a JSON or YAML document. These templates describe automated workflow configurations for conversational chat or query generation, AI connectors, tools, agents, and other components that prepare OpenSearch as a backend for generative models. For template examples, see [Sample templates](https://github.com/opensearch-project/flow-framework/tree/main/sample-templates).
+OpenSearch use case templates provide a compact description of the setup process in a JSON or YAML document. These templates describe automated workflow configurations for conversational chat or query generation, AI connectors, tools, agents, and other components that prepare OpenSearch as a backend for generative models. For custom template examples, see [Sample templates](https://github.com/opensearch-project/flow-framework/tree/main/sample-templates). For OpenSearch-provided templates, see [Workflow templates]({{site.url}}{{site.baseurl}}/automating-configurations/workflow-templates/).
```
Comment (vagimeli) [inline _automating-configurations/index.md:19]: Should AI be defined on first mention?

**Review by vagimeli — APPROVED** [review]
LGTM. See my one comment.

File: _automating-configurations/index.md (hunk 4 above)
Comment (kolchfa-aws) [reply _automating-configurations/index.md:19]: Not anymore. From [TERMS](https://github.com/opensearch-project/documentation-website/blob/main/TERMS.md):

AI

No need to define as artificial intelligence (AI).

**Pull request merged**